src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -114,9 +114,11 @@ internal void EnsureInitialized(WebHostSettings settings)
             if (!WebScriptHostManager.InStandbyMode)
             {
                 // standby mode can only change from true to false
-                // When standby mode changes, we reset all instances
+                // when standby mode changes, we reset all instances
                 if (_activeHostManager == null)
                 {
+                    _settingsManager.Reset();
```

Comment:
I think we should be resetting this earlier - before we set _activeHostManager, otherwise the potential exists for the old pre-specialization settings to be used.